src/agent/LibFuzzerDotnetLoader/Program.cs

```diff
@@ -81,4 +70,107 @@ static bool TryTestOne<T>(MethodInfo method, Func<T, SharpFuzz.ReadOnlySpanActio
             return TryTestOne<TestOneArray>(method, t => span => t(span.ToArray()));
         }
     }
+
+    class LibFuzzerDotnetTarget
+    {
+        public string AssemblyPath { get; private set; }
+        public string ClassName { get; private set; }
+        public string MethodName { get; private set; }
+
+        public LibFuzzerDotnetTarget(string assemblyPath, string className, string methodName)
+        {
+            AssemblyPath = assemblyPath;
+            ClassName = className;
+            MethodName = methodName;
+        }
+
+        public static LibFuzzerDotnetTarget FromEnvironment()
+        {
+            try {
+                return FromEnvironmentVarDelimited();
+            }
+            catch
+            {}
+
+            try {
+                return FromEnvironmentVars();
+            }
+            catch
+            {}
+
+            throw new Exception("No fuzzing target specified by environment variables");
+        }
+
+        static LibFuzzerDotnetTarget FromEnvironmentVars()
+        {
+            var assemblyPath = Environment.GetEnvironmentVariable("LIBFUZZER_DOTNET_TARGET_ASSEMBLY");
+
+            if (assemblyPath is null)
+            {
+                throw new Exception("`LIBFUZZER_DOTNET_TARGET_ASSEMBLY` not set");
+            }
+
+            var className = Environment.GetEnvironmentVariable("LIBFUZZER_DOTNET_TARGET_CLASS");
+
+            if (className is null)
+            {
+                throw new Exception("`LIBFUZZER_DOTNET_TARGET_CLASS` not set");
+            }
+
+            var methodName = Environment.GetEnvironmentVariable("LIBFUZZER_DOTNET_TARGET_METHOD");
+
+            if (methodName is null)
+            {
+                throw new Exception("`LIBFUZZER_DOTNET_TARGET_METHOD` not set");
+            }
+
+            return new LibFuzzerDotnetTarget(assemblyPath, className, methodName);
+        }
+
+
+        static LibFuzzerDotnetTarget FromEnvironmentVarDelimited()
+        {
+            string? assemblyPath = null;
+            string? className = null;
+            string? methodName = null;
+
+            var target = Environment.GetEnvironmentVariable("LIBFUZZER_DOTNET_TARGET");
+
+            if (target is null)
+            {
+                throw new Exception("`LIBFUZZER_DOTNET_TARGET` not set. " +
+                                    "Expected format: \"<assembly-path>:<class>:<static-method>\"");
+            }
+
+            var parts = target.Split(':');
```

Comment:
you can check that parts.Length == 3. that will save you the other checks.
You should also add StringSplitOptions.RemoveEmptyEntries

Reply:
This worked out really well. I kind of wanted the component-wise checks for specificity, but their value is questionable if the env var is wrong, especially given the use of split options.